src/console/ipc.ts

```diff
@@ -0,0 +1,18 @@
+import { _, EmptyPayload, makeEndpoint, SuccessPayload } from "../ipc";
+import { DiscoveredConsoleInfo } from "./discovery";
+import { MirrorDetails } from "./types";
+
+export const addMirrorConfig = makeEndpoint.main("addMirrorConfig", <{ config: MirrorDetails }>_, <SuccessPayload>_);
```

Comment:
It's not good that `MirrorDetails` is accepted as the input parameters, since `MirrorDetails` is not entirely JSON serializable. We should only pass serializable JSON through IPC.

As a reminder, this is the type of `MirrorDetails`.

```ts
export interface MirrorDetails {
  id?: number;
  ipAddress: string;
  port: number;
  folderPath: string;
  isRealTimeMode: boolean;
  isRelaying?: boolean;
  connection?: ConsoleConnection | null;
  fileWriter?: SlpFileWriter | null;
  isMirroring?: boolean;
  obsSettings?: OBSSettings | null;
  obsManager?: OBSManager | null;
}
```

We should be more explicit in what values are allowed in the config, since we never expect the frontend to send things like `OBSManager` and `SlpFileWriter`s through IPC to the main.